pyqtgraph/debug.py

```diff
@@ -1151,35 +1155,51 @@ def start(self, interval=None):
         self.thread.start()
 
     def run(self):
-        while True:
-            with self.lock:
-                if self._stop is True:
-                    return
-                    
-            print("\n=============  THREAD FRAMES:  ================")
-            for id, frame in sys._current_frames().items():
-                if id == threading.current_thread().ident:
-                    continue
+        iter = 0
+        if self.logFile is None:
```

Comment:
sys.stdout behaves like a file object, so we should be able to do something like....

```python
printFile = sys.stdout if self.logFile is None else self.logFile
```

we can nick all the `if self.logFile` checks and then use `printFile.write` everywhere.